### PR TITLE
remove cov-core, nose-cov and fix their dependencies

### PR DESCRIPTION
**Pull request by narang99**
## Description of changes
was meant to be a patch for: https://github.com/NixOS/nixpkgs/issues/245383   

- removed cov-core since its outdated
- removed nose-cov since its outdated
- removed them as deps in dyn and pylti

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Extra stuff:
- Tested by installing both cov-core and covCore in local nix-env using my local repository
- Ran this command `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`
   - got inside a nix shell
   - ran python and `import cov_core`
   - `core_core.__file__` looks like this: `/nix/store/2g0dkhywpawpnbqw5aijxpik6si3wq4f-python3.11-cov-core-1.15.0/lib/python3.11/site-packages/cov_core.py`

### CODE REVIEW

**Comment by narang99** [comment]
I had created the PR as draft and it did not ping anyone, can you help me here @mweinelt ? I don't know who to tag :)

**Comment by natsukium** [comment]
Thanks for your contributions! The change looks good to me.
However, this package has been unmaintained for almost 10 years and archived, so it seems like a good idea to consider removing it.
https://github.com/schlamar/cov-core

Finally, I would appreciate it if you could read the contribution guides and change the commit message.
https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines-contributing-guidelines

**Comment by narang99** [comment]
I saw the usages of cov-core to remove it and the following packages use it:
- pylti (commit is old, but it doesn't look like we'll take this out). Last commit: 2020
- nose-cov (the last commit's 2012, accessing its source page on PyPI shows 404, definitely can remove it)
- dyn (this has a recent last release [2023])

I can start taking out `nose-cov` and try removing `cov-core` refs in the other two, but can you help me understand how I can locally test it out? Just wanna know what unit tests I should run, don't wanna run the whole suite as it looks big

**Comment by natsukium** [comment]
Thanks for looking into this.
It looks like `nose-cov` can be easily removed as you said.
As for the other two, I looked at the source code and they don't seem to need `cov-core` anymore.
In the first place, we don't need it because `cov-core` is a package for testing and we don't need to measure coverage in the nixpkgs check phase.
So, for these two packages, you can just remove `cov-core` from the input.

Since this change is not large, you can easily check all changes using [nixpkgs-review](https://github.com/Mic92/nixpkgs-review).

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/development/python-modules/pylti/default.nix [hunk 1]
```diff
@@ -12,7 +12,6 @@
 , pytest
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pylti/default.nix:12]: This package is 2.7 only and has no reverse deps. We could also remove it.

**Comment by natsukium** [comment]
done in other PRs

**Pull request closed without merging**